### ISSUE TITLE
support object without properties 

### ISSUE THREAD

**Issue by FalcoSuessgott** [dikhan/terraform-provider-openapi#292]
## Is your feature request related to a problem?
We want to use that provider as a netbox terraform provider, which supports OpenAPIv2.

Currently defintions of type objects without any properties are not supported.
Netbox offers the possibility to add custom fields to objects. Those custom fields can have various data types (bool, string, date, ...).

Since in the OpenAPI spec those custom_fields are represented as objects without any properties: 
```json
    "Secret": {
      "required": [
        "assigned_object_type",
        "assigned_object_id",
        "role",
        "plaintext"
      ],
      "type": "object",
      "properties": {
        "id": {
          "title": "ID",
          "type": "integer",
          "readOnly": true
        },
        [....]
        "custom_fields": {
          "title": "Custom fields",
          "type": "object",
          "default": {}
        },
```

Currently the provider responds the following:
```sh
│ 23 errors occurred:
   ....
│ 	* resource netbox_secrets: custom_fields: Default is not valid for lists or sets
```

As a terraform schema, `custom_fields` are represented as the following
```hcl
"custom_fields": {
  Type:     schema.TypeMap,
  Optional: true,
},
```

## Describe the solution you'd like
It would be nice to have maps supported or to find a workaround to get that provider working with empty objects.

## Describe alternatives you've considered
Maybe you can give us some hints where to start implementing this feature on our own.

## Checklist (for admin only)

**Don't forget to go through the checklist to make sure the issue is created properly**:

- [ ] I have added a corresponding label (feature request) to the issue (right side menu)
- [ ] I have added this issue to the 'API Terraform Provider' GitHub project (right side menu)

**Comment by dikhan** [comment]
Hi @FalcoSuessgott ,

Thanks for opening the issue, I was just reading the [custom field docs](https://netbox.readthedocs.io/en/stable/additional-features/custom-fields/#custom-fields-and-the-rest-api) to try and get a bit more context and was wondering about the following:

-  If you have control over the API code, have you considered defining the model for the custom fields as follows? This would provide the flexibility to users to provide custom fields knowing the expected types supported and the backend could perform the corresponding validation upon receipt (eg: ensuring a integer type custom field value is actually an integer, etc). This is already supported by the OpenAPI Terraform provider, so if you have any control over the API that could be something to try out.

````
definitions:
  Secret:
    type: "object"
    required:
      ...
    properties:
...
      custom_fields:
        title: "Custom fields"
        type: "array"
        items:
          type: "object"
          properties:
            type:
              description: "custom property type, must be one of the supported types as specified in the regex"
              regex: "string|integer|boolean|date|url|selection|multiple_selection"
              type: string
            value:
              type: string
````

- Alternatively, this could also be tackled by adding support for map[string]string into the OpenAPI Terraform provider as documented in #80 . This would seem to cover the examples in the [Custom Fields and the REST API](https://netbox.readthedocs.io/en/stable/additional-features/custom-fields/#custom-fields-and-the-rest-api#custom-fields-and-the-rest-api):

````
{
    "id": 123,
    "url": "http://localhost:8000/api/dcim/sites/123/",
    "name": "Raleigh 42",
    ...
    "custom_fields": {
        "deployed": "2018-06-19",
        "site_code": "US-NC-RAL42"
    },
    ...
````

Hope that helps!

Thanks
Dani

**Comment by dikhan** [comment]
Hi @FalcoSuessgott ,

Just wanted to follow up on this and see whether the previous response help at all. Reading again the error you got from the terraform provider I wonder if removing the Default from the OpenApi spec for the object property might also unblock you:

```
  "Secret": {
      "required": [
        "assigned_object_type",
        "assigned_object_id",
        "role",
        "plaintext"
      ],
      "type": "object",
      "properties": {
        "id": {
          "title": "ID",
          "type": "integer",
          "readOnly": true
        },
        [....]
        "custom_fields": {
          "title": "Custom fields",
          "type": "object",
        },
```

Thanks,
Dani
